File: python/rules/testing/no-test-case-ordering.py
import pytest

# ruleid: python-no-test-case-ordering
@pytest.mark.order(1)
def test_first_step():
    assert 1 == 1

# ruleid: python-no-test-case-ordering
@pytest.mark.run(order=2)
def test_second_step():
    assert 2 == 2

# ok: python-no-test-case-ordering
def test_independent_step_a():
    assert 1 == 1

# ok: python-no-test-case-ordering
def test_independent_step_b():
    assert 2 == 2

File: csharp/rules/testing/no-test-case-ordering.cs
using Xunit;

namespace Thruput.Semgrep.Tests
{
    // ruleid: csharp-no-test-case-ordering
    [TestCaseOrderer("MyOrderer", "MyAssembly")]
    public class OrderedTestSuite
    {
        [Fact]
        // ruleid: csharp-no-test-case-ordering
        [TestPriority(1)]
        public void StepOne()
        {
            Assert.True(true);
        }

        [Fact]
        // ruleid: csharp-no-test-case-ordering
        [Order(2)]
        public void StepTwo()
        {
            Assert.True(true);
        }
    }

    public class IndependentTestSuite
    {
        // ok: csharp-no-test-case-ordering
        [Fact]
        public void IndependentTestA()
        {
            Assert.Equal(1, 1);
        }

        // ok: csharp-no-test-case-ordering
        [Fact]
        public void IndependentTestB()
        {
            Assert.Equal(2, 2);
        }
    }

    public class TestPriorityAttribute : System.Attribute
    {
        public TestPriorityAttribute(int priority) { }
    }

    public class OrderAttribute : System.Attribute
    {
        public OrderAttribute(int order) { }
    }
}
